Multisig/UI/WalletConnect/WalletConnectClientController.swift

```diff
@@ -157,6 +127,53 @@ class WalletConnectClientController {
         }
     }
 
+    private func sign(transaction: Transaction, completion: @escaping (Result<String, Error>) -> Void) {
+        guard let session = session,
+              let client = client,
+              let walletAddress = session.walletInfo?.accounts.first else {
+            completion(.failure(GSError.WalletNotConnected(description: "Could not sign transaction")))
+            return
+        }
+
+        func handleResponse(_ response: Response) {
+            do {
+                var signature = try response.result(as: String.self)
+
+                var signatureBytes = Data(hex: signature).bytes
+                var v = signatureBytes.last!
+                if v < 27 {
+                    v += 27
+                    signatureBytes[signatureBytes.count - 1] = v
+                    signature = Data(signatureBytes).toHexStringWithPrefix()
+                }
+
+                completion(.success(signature))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+
+        do {
+            switch session.walletInfo?.peerMeta.name ?? "" {
+
+            // we call signTypedData only for wallets supporting this feature
+            case "MetaMask", "LedgerLive", "🌈 Rainbow", "Trust Wallet":
```

Comment:
what if their name changes? do they have some id?

Reply:
We have only peerMeta like 
```
"peerMeta":{"description":"MetaMask Mobile app","icons":["https:\/\/raw.githubusercontent.com\/MetaMask\/brand-resources\/master\/SVG\/metamask-fox.svg"],"name":"MetaMask","ssl":true,"url":"https:\/\/metamask.io"},"rpcUrl":""}}
```

So I thought that name is quite reliable. Otherwise eth_sign will be used that is also supported by majority of wallets, but not as user friendly as eth_signTypedData